libopenage/util/fixed_point.h

```diff
@@ -446,8 +447,37 @@ class FixedPoint {
 		return is;
 	}
 
+	/**
+	 * Pure FixedPoint sqrt implementation using Heron's Algorithm.
+	 *
+	 * Note that this function is undefined for negative values.
+	 */
 	constexpr double sqrt() {
```

Comment:
It's probably better that we make the return type `FixedPoint` by default. If we want a floating point sqrt, there's already the std::sqrt overload in line 606.